codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoSettings.java

```diff
@@ -47,13 +54,16 @@ public final class GoSettings {
      */
     public static GoSettings from(ObjectNode config) {
         GoSettings settings = new GoSettings();
-        config.warnIfAdditionalProperties(Arrays.asList(SERVICE, MODULE_NAME, MODULE_DESCRIPTION));
+        config.warnIfAdditionalProperties(
+            Arrays.asList(SERVICE, MODULE_NAME, MODULE_DESCRIPTION, MODULE_VERSION, GENERATE_GO_MOD, GO_DIRECTIVE));
 
         settings.setService(config.expectStringMember(SERVICE).expectShapeId());
         settings.setModuleName(config.expectStringMember(MODULE_NAME).getValue());
         settings.setModuleDescription(config.getStringMemberOrDefault(
                 MODULE_DESCRIPTION, settings.getModuleName() + " client"));
-
+        settings.setModuleVersion(config.getStringMemberOrDefault(MODULE_VERSION, null));
+        settings.setGenerateGoMod(config.getBooleanMemberOrDefault(GENERATE_GO_MOD, false));
+        settings.setGoDirective(config.getStringMemberOrDefault(GO_DIRECTIVE, "1.15"));
```

Comment:
These settings could reference static fields instead of hardcoding / duplicating the default values.